### DISCUSSION TITLE
incremental attitude control

### DISCUSSION THREAD

**Discussion by adarsh2798** [bitcraze/discussions#1336] [Q&A]
Hi,

I need to do incremental attitude control of crazyflie. Meaning, whatever attitude the crazyflie currently is in, from there I want it to do some $\Delta$ pitch, $\Delta$ roll maneuver. For example if crazyflie is at say 15deg pitch. From that attitude, i want it to do $\Delta$ pitch=2 degrees maneuver. After this maneuver the new pitch would be 17deg, but instead of sending the final pitch i want to send incremental commands every iteration. 
What function shall I use in cflib for this?

Thanks!

**Comment by knmcguire** [comment]
Currently the closest that comes to this is doing RPYT rate control. That is bascially using the send_setpoint() and setting a parameters before that.

```
    roll    = 0.0
    pitch   = 0.0
    yawrate = 0
    thrust  = 10001
    crazyflie.commander.send_setpoint(roll, pitch, yawrate, thrust)

And mke sure make sure that the [flightmode.stabModeRoll/Pitch/Yaw](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#flightmode) is set to rate (0).
```

Other solutions is to setup a log callback on the attitude and combine that with the absolute attitude setpoint send perhaps. You can find instructions on how to do that in the cflib here: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/python_api/
